android/src/main/kotlin/io/piano/flutter/piano_analytics/PianoAnalyticsPlugin.kt

```diff
@@ -97,10 +118,19 @@ class PianoAnalyticsPlugin : FlutterPlugin, MethodCallHandler, ActivityAware {
             configuration = Configuration.Builder(
                 site = call.arg("site"),
                 collectDomain = call.arg("collectDomain"),
-                visitorIDType = getVisitorIDType(call.arg("visitorIDType"))
+                visitorIDType = getVisitorIDType(call.arg("visitorIDType")),
+                visitorStorageLifetime = call.argument<Int>("storageLifetimeVisitor")
+                    ?: Configuration.DEFAULT_VISITOR_STORAGE_LIFETIME,
+                visitorStorageMode = getVisitorStorageMode(call.argument("visitorStorageMode")),
+                ignoreLimitedAdTracking = call.argument<Boolean>("ignoreLimitedAdvertisingTracking")
+                    ?: false
             ).build(),
             pianoConsents = getPianoConsents()
         )
+        val visitorId = call.argument<String>("visitorId")
+        if (visitorId != null) {
```

Comment:
There should be a check for visitorIDType == custom (or forcing custom type)

Reply:
added check